rectools/models/rank.py

```diff
@@ -191,15 +203,47 @@ def rank(
 
         real_k = min(k, object_factors.shape[0])
 
-        ids, scores = implicit.cpu.topk.topk(  # pylint: disable=c-extension-no-member
-            items=object_factors,
-            query=subject_factors,
-            k=real_k,
-            item_norms=object_norms,  # query norms for COSINE distance are applied afterwards
-            filter_query_items=filter_query_items,  # queries x objects csr matrix for getting neginf scores
-            filter_items=None,  # rectools doesn't support blacklist for now
-            num_threads=num_threads,
-        )
+        if use_gpu and not HAS_CUDA:
+            warnings.warn("Forced rank() on CPU")
+            use_gpu = False
+
+        if use_gpu:
```

Comment:
```suggestion
        if use_gpu:  # pragma: no cover
```
To stop coverage from failing since we don't have gpu for tests on github.

Also please move all gpu logic to a separate private method. It makes sense and linters will stop complaining.

We merge as soon as all tests pass.

Reply:
I hope now it will run without any problem. Also I noticed that I forgot to add tests for sparse subject factors, fixed.

Reply:
Please add `# pragma: no cover` comment to ignore gpu logic for coverage test.

Reply:
Done